spec/src/language_negotiation_lambda_spec.js

```diff
@@ -0,0 +1,143 @@
+                  "distributionId": "EXAMPLE"
+                }
+              }
+            }
+        ]
+    };
+    var default_header = ['en'];
+
+    beforeEach(function(){
+        delete event.Records[0].cf.request.headers["Accept-Language"];
+        delete event.Records[0].cf.request.headers["X-Accept-Language"];
+    });
+    describe( 'Well formed headers', function(){
+        [
+            { header: "en",                   customHeader: "en" },
+            { header: "es",                   customHeader: "es" },
+            { header: "de",                   customHeader: "en" },
+            { header: "en;q=0.8,es",          customHeader: "es" },
+            { header: "es;q=0.8,en",          customHeader: "en" },
+            { header: "de,en;q=0.8",          customHeader: "en" },
+            { header: "de,es;q=0.8",          customHeader: "es" },
+            { header: "es;q=NaN",             customHeader: "es" },
+            { header: "es;q=NaN,en;q=0.01",   customHeader: "en" },
+            { header: "de,en;q=0.2,es;q=0.8", customHeader: "es" },
+            { header: "EN",                   customHeader: "en" },
+            { header: "ES",                   customHeader: "es" },
+            { header: "EN-US",                customHeader: "en" },
+            { header: "ES-419",               customHeader: "es" }
+        ].forEach( function( headers ) {
+            it( `Resolves header ${headers.header} to ${headers.customHeader}`, function( done ) {
+                event.Records[0].cf.request.headers['Accept-Language'] = [headers.header];
+
+                languageNegotiationLambda.handler( event, { /* context */ }, (err, result) => {
+                    try {
+                        expect( err ).toBeNull();
+                        expect( result.headers['X-Accept-Language'] ).toEqual([headers.customHeader]);
+                        done();
+                    }
+                    catch( error ) {
+                        done( error );
+                    }
+                });
+            });
+        });
+    });
+
+
+    describe( 'malformed headers', function(){
+        [
+            "",
+            "dhbeiyu292dfiue2",
+            undefined,
+            NaN,
+            null
+        ].forEach( function( header ) {
+            it( `Resolves header ${header} to default`, function( done ) {
+                event.Records[0].cf.request.headers['Accept-Language'] = [header];
+
+                languageNegotiationLambda.handler( event, { /* context */ }, (err, result) => {
+                    try {
+                        expect( err ).toBeNull();
+                        expect( result.headers['X-Accept-Language'] ).toEqual(default_header);
+                        done();
+                    }
+                    catch( error ) {
+                        done( error );
+                    }
+                });
+            });
+        });
+    });
+
+    describe( 'Errors', function(){
```

Comment:
Could we just move the console.log checks into the test on line 83 and remove the rest of this block? I realize that doing so might make it difficult to detect the exact error message, but I believe you can check that .callArgs[0] includes the string 'Error'.